gui/wxpython/datacatalog/catalog.py

```diff
@@ -39,9 +39,11 @@ def __init__(self, parent, giface=None, id=wx.ID_ANY,
                  title=_("Data catalog"), name='catalog', **kwargs):
         """Panel constructor  """
         self.showNotification = Signal('DataCatalog.showNotification')
+        self.showPage = Signal('DataCatalog.showPage')
```

Comment:
The signals serve us in two ways, they remove direct code dependence and also remove conceptual dependence. This one removes the code dependence, but does not address the conceptual dependence. The show notification signal above just says, "Hi there, I have a some sort of notification, so dear class user, if you think the end user wants to see them, here is one you may want to show now." The show page signal, on the other hand, says "Hi there, switch the page if you can, I have no idea what it is, but try to switch it anyway."

In other words, the signal (name, parameters) should be relevant to the object emitting the signals. `DataCatalog.showNotification` says to show notification generated by the data catalog. `DataCatalog.showPage` makes assumptions about surrounding code and imports this assumption into its API resulting in an some _page_ being part of data catalog.

One test of this is the standalone data catalog (g.gui...). Obviously, I don't have any page there to show and ignoring a signal is fine (at least unless otherwise noted). I can choose to ignore even the notification signal, but there it is clear what I'm ignoring. With the show page signal, it is not clear what event I'm ignoring and what should I do in case I want to react to what happened in the data catalog.